electricitymap/contrib/lib/models/event_lists.py

```diff
@@ -69,8 +76,50 @@ def append(
         if event:
             self.events.append(event)
 
+    @staticmethod
+    def merge_production_breakdowns(
+        ungrouped_production_breakdowns: List["ProductionBreakdownList"],
+        logger: Logger,
+        merge_zone_key: ZoneKey,
+        merge_source: str,
+    ) -> "ProductionBreakdownList":
+        """Given multiple parser outputs, sum the production and storage of corresponding datetimes to create a unique production breakdown list."""
+        if len(ungrouped_production_breakdowns) == 0:
+            return ProductionBreakdownList(logger)
+
+        # Create a dataframe for each parser output, then flatten the power mixes.
+        prod_and_storage_dfs = [
+            pd.json_normalize(breakdowns.to_list()).set_index("datetime")
+            for breakdowns in ungrouped_production_breakdowns
+        ]
+        df = pd.concat(prod_and_storage_dfs)
+        df = df.groupby(level=0, dropna=False).sum(numeric_only=True, min_count=1)
+        result = ProductionBreakdownList(logger)
+        for row in df.iterrows():
+            production_mix = ProductionMix()
+            storage_mix = StorageMix()
+            for key, value in row[1].items():
+                if np.isnan(value):
+                    value = None
+                # The key is in the form of "production.<mode>" or "storage.<mode>"
+                prefix, mode = key.split(".")
```

Comment:
key does not hava a defined type so the type checker is trhowing an error here that split does not exist on the type hashable.

Reply:
I don't think there's an easy way out of this at the moment so I just added a type ignore.